### PR TITLE
initial implementation of celeb blushlist

### PR DESCRIPTION
**Pull request by mozkeeler**
So, this isn't scalable at all, but it'll work as a first pass. I just used the Forbes list of top 100 celebrities for 2012 to get some names.

### CODE REVIEW

**Comment by mozkeeler** [comment]
It turns out, we need to whitelist search terms and deliberately not whitelist a domain when a user says "open normally" for a search term blush interception. We may want to move to a more general whitelist mechanism, but for now the duplication of functionality should work. Also, we'll probably want to do something more clever about identifying blushy search terms instead of what is basically simple lowercase string matching.

**Comment by monicachew** [comment]
Yes, I expected that. We are going to have to treat search differently always in both the matching case and the whitelist case.

File: lib/celebs.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+// This file is automatically generated by make_celebs.py
+let celebs = {
```
Comment (monicachew) [inline lib/celebs.js]: We need a more generic name here. I think this is more like queries.js, or searchterms.js.

File: lib/bpContentPolicy.js [hunk 2]
```diff
@@ -68,7 +93,8 @@ exports.bpContentPolicy = Class({
         let query = bpUtil.getSearchTermFromURI(aContentLocation);
         if ((bpCategorizer.getCategoryForBlushlist(aContentLocation.host) ||
              bpCategorizer.isQueryEmbarrassing(query)) &&
```
Comment (monicachew) [inline lib/bpContentPolicy.js:95]: I don't think this quite works. getSearchTermFromURI returns the urlencoded string which can include %20 or + for space. We need to urldecode the string.

File: lib/bpContentPolicy.js [hunk 3]
```diff
@@ -22,6 +22,15 @@ if (!ss.storage.whitelistedDomains) {
 }
 
 /**
+ * A map of string -> bool indicating that searches with the given query
+ * string should not be opened in a private window. This persists across
+ * restarts.
+ */
+if (!ss.storage.whitelistedQueries) {
```
Comment (monicachew) [inline lib/bpContentPolicy.js:29]: You don't have to change this now, but our simple storage object is getting a little messy. I propose moving the initialization all to one spot (bpUtil?) and making a top level simple storage object. I'll file a bug.

File: lib/celebs.js (hunk 1 above)
Comment (monicachew) [inline lib/celebs.js]: It should probably also contain some adult search terms, which leads me to believe that it should probably mirror the structure of blushlist.js, where the map value is the category.

File: util/make_celebs.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+try:
+  f_in = open(sys.argv[1], "r")
+  f_out = open("celebs.js", "a")
+except:
+  sys.exit("Can't find file")
+
+f_out.write("// This file is automatically generated by make_celebs.py\n")
+f_out.write("let celebs = {\n")
+for l in f_in.readlines():
+  l = l.strip().lower().replace("\"", "")
```
Comment (monicachew) [inline util/make_celebs.py]: What is this call to replace doing? I don't see any " in your input file.

File: util/make_celebs.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/python
+# Create celebs.js from a list of celebrities
+# Usage: make_celebs.py <input_file>
```
Comment (monicachew) [inline util/make_celebs.py]: Same with the naming here. make_queries or make_searchterms.py?

**Comment by monicachew** [comment]
I think this is really close. For future-proofing, I think that keeping query terms and urls on separate maps makes more sense. URLs and search terms are semantically different, and I can see bugs where "whitehouse.com", "whitehouse.gov", and search query "whitehouse" need to be treated differently, but because search queries are conflated with URLs in the current PR it's easy to make a programming error.

One easy fix is just to make make_blushlist.py take an output param, then you can just run the same script with two different output params.

File: lib/celebs.js (hunk 1 above)
Comment (monicachew) [inline lib/celebs.js]: I'm sorry if this comment was confusing. I should have said to keep a separate map.

**Pull request merged**